### DISCUSSION TITLE
Are stateful services encrypted?

### DISCUSSION THREAD

**Discussion by TomasMalcanek** [shinyorg/shiny#1467] [Q&A] [answered]
I am using stateful services and I have a question, whether the stateful services are saved into the SecureStorage by default, or just in plain text somewhere.

I am asking this in terms of security storing auth tokens and perhaps stored user's login password.

**Comment by aritchie** [comment, marked as answer]
They can be stored in secure storage if you want

```csharp
[ObjectBinder("secure")]
public interface IWhatever : INotifyPropertyChanged ....
```

Otherwise they go to the default store on the platform which is settings for both Android & iOS

**Reply by TomasMalcanek** [reply]
So I've tried it now, and there is no `ObjectBinder` class.
After searching, the closest I found is `Shiny.Stores.ObjectStoreBinder`, is that correct?

```csharp
[ObjectStoreBinder("secure")]
public interface IWhatever : INotifyPropertyChanged ....
```

Documentation I found:
https://github.com/shinyorg/shinyorg.github.io/blob/6139ea086a8afa7477ed978bda96e579d68d3df3/mobile/other/keyvalue.html#L345-L350
